sklearn/ensemble/bagging.py

```diff
@@ -387,6 +397,56 @@ def _validate_y(self, y):
         # Default implementation
         return column_or_1d(y, warn=True)
 
+    def _get_estimators_data_draws(self, sample_mask=False, feature_mask=False):
+        # If max_samples is float
+        if not isinstance(self.max_samples, (numbers.Integral, np.integer)):
+            max_samples = int(self.max_samples * self._n_samples)
+
+        if not (0 < max_samples <= self._n_samples):
+            raise ValueError("max_samples must be in (0, n_samples]")
+
+        # If max_features is float
+        if not isinstance(self.max_features, (numbers.Integral, np.integer)):
+            max_features = int(self.max_features * self.n_features_)
+
+        if not (0 < max_features <= self.n_features_):
+            raise ValueError("max_features must be in (0, n_samples]")
+
+        # Get sampled indices or mask
+        for seed in self._seeds:
+            # Operations accessing random_state must be performed identically
+            # to those in `_parallel_build_estimators()`
+            random_state = check_random_state(seed)
+            seed = random_state.randint(MAX_INT) 
+            feature_indices, sample_indices = _generate_bagging_indices(random_state, 
+                                                                  self.bootstrap_features,
+                                                                  self.bootstrap, 
+                                                                  self.n_features_, 
+                                                                  self._n_samples, 
+                                                                  max_features, 
+                                                                  max_samples)
+
+            # Convert feature indices to mask if desired
+            if feature_mask:
```

Comment:
don't we have a helper function for this? I guess we have ``save_mask`` which does the opposite. this seems like such a common pattern..